cmd/main.go

```diff
@@ -96,9 +98,18 @@ func main() {
 		logger.Fatalw("failed to create kubernetes client", err)
 	}
 
+	// Load Config For the pod
+	configManager := config.NewConfigManager()
+	err = configManager.LoadConfig(func(err error) {
+		logger.Errorw("Failed to reload global configuration file", err)
+	}, configPath)
+	if err != nil {
+		logger.Fatalw("Failed to load config file", err)
+	}
 	reconciler, err := controller.NewGitSyncReconciler(
 		kubeClient,
 		mgr.GetScheme(),
+		configManager.GetConfig(),
```

Comment:
let's pass the `configManger` in instead. That should be what gets passed in everywhere. Then, the user who needs updated configuration can always call `configManager.GetConfig()` whenever it needs an atomically correct version of the configuration.